src/handlers.rs

```diff
@@ -134,6 +142,125 @@ pub fn handle_request(
             Ok(vec![])
         }
 
+        RequestType::GETAI => {
+            let hostname = CStr::from_bytes_with_nul(request.key)?.to_str()?;
+            // Boths hints are necessary to mimick the glibc behaviour.
+            let hints = AddrInfoHints {
+                // The canonical name will be filled in the first
+                // addrinfo struct returned by getaddrinfo.
+                flags: AI_CANONNAME,
+                // Deduplicate the resulting addrs.
+                socktype: SOCK_STREAM,
```

Comment:
Why `SOCK_STREAM` here? I guess, what if the nscd client is trying to do `SOCK_DGRAM`, what do we do and what would real `nscd` do?

Reply:
Added

```rust
                // There's no way to convey socktype in the the Nscd
                // protocol, neither in the request nor response.
                //
                // Set this to SOCK_STREAM to match glibc and unscd
                // behaviour.
```

Does it make sense?

Reply:
@leifwalsh please mark as resolved if you're ok with this.